test_package_dag_filter sometimes fails

Testing of pipdeptree 2.9.4 revealed the `test_package_dag_filter` test sometimes fails.  The test usually pass, but about every 10th test run shows failure like these: 
```
___________________________ test_package_dag_filter ____________________________

    def test_package_dag_filter():
        # When both show_only and exclude are not specified, same tree
        assert t.filter_nodes(None, None) is t

        # when show_only is specified
        g1 = dag_to_dict(t.filter_nodes({"a", "d"}, None))
        expected = {"a": ["b", "c"], "b": ["d"], "c": ["d", "e"], "d": ["e"], "e": []}
>       assert expected == g1
E       AssertionError: assert {'a': ['b', '...': ['e'], ...} == {'a': ['b', '...': ['e'], ...}
E         Omitting 4 identical items, use -vv to show
E         Differing items:
E         {'c': ['d', 'e']} != {'c': ['e', 'd']}
E         Full diff:
E         - {'a': ['b', 'c'], 'b': ['d'], 'c': ['e', 'd'], 'd': ['e'], 'e': []}
E         ?                                        -----
E         + {'a': ['b', 'c'], 'b': ['d'], 'c': ['d', 'e'], 'd': ['e'], 'e': []}
E         ?                                     +++++

tests/test_pipdeptree.py:93: AssertionError
```
or
```
___________________________ test_package_dag_filter ____________________________ 
  
    def test_package_dag_filter():
        # When both show_only and exclude are not specified, same tree
        assert t.filter_nodes(None, None) is t
    
        # when show_only is specified
        g1 = dag_to_dict(t.filter_nodes({"a", "d"}, None))
        expected = {"a": ["b", "c"], "b": ["d"], "c": ["d", "e"], "d": ["e"], "e": []}
>       assert expected == g1
E       AssertionError: assert {'a': ['b', '...': ['e'], ...} == {'a': ['c', '...': ['e'], ...}
E         Omitting 3 identical items, use -vv to show
E         Differing items:
E         {'a': ['b', 'c']} != {'a': ['c', 'b']}
E         {'c': ['d', 'e']} != {'c': ['e', 'd']}
E         Full diff:
E         - {'a': ['c', 'b'], 'b': ['d'], 'c': ['e', 'd'], 'd': ['e'], 'e': []}
E         ?           -----                        -----
E         + {'a': ['b', 'c'], 'b': ['d'], 'c': ['d', 'e'], 'd': ['e'], 'e': []}
E         ?        +++++                        +++++

tests/test_pipdeptree.py:93: AssertionError
```
I run tests with the `randomly` pytest plugin installed so maybe it is related.  Please note that this is the only test I see failed.

--- a/tests/test_pipdeptree.py
+++ b/tests/test_pipdeptree.py
@@ -774,7 +774,7 @@ def randomized_dag_copy(raw: PackageDAG) -> PackageDAG:
     randomized_nodes = list(raw._obj.keys())  # noqa: SLF001
     random.shuffle(randomized_nodes)
     for node in randomized_nodes:
-        edges = raw._obj[node]  # noqa: SLF001
+        edges = raw._obj[node].copy()  # noqa: SLF001
         random.shuffle(edges)
         randomized_graph[node] = edges
     assert set(randomized_graph) == set(raw._obj)  # noqa: SLF001

